### PR TITLE
some dashboards enhancements

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

- [x] new widget: summary 
image

- [x] new widget: articles list (for rss and reminders)
image

- [x] point labels 
    * [x] on line/area
    image

    * [x] on bars
    image

    * [x] add option in card to switch the feature
    image

- [x] remove 0 value points for line/area graphs
   image

- [x] replace svg points by circle (to have a cutted effect) on line/area graph
   image

- [x] minimum size for widget (to avoid too small widgets at creation)

- [x] filters

image
image

- [x] need to check #7785 and #7957

### CODE REVIEW

File: inc/dashboard/filter.class.php [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    * Return all available filters
+    * Plugins can hooks on this functions to add their own filters
+    *
+    * @return array of filters
+    */
+   static function getAll(): array {
+      $filters = [
+         'dates'        => __("Creation date"),
+         'dates_mod'    => __("Last update"),
+         'itilcategory' => \ITILCategory::getTypeName(\Session::getPluralNumber()),
```
Comment (trasher) [inline inc/dashboard/filter.class.php:54]: master branch requires global declaration of uses at the top of the file

File: inc/dashboard/filter.class.php [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    * Get HTML for a dates range filter
+    *
+    * @param string|array $values init the input with these values, will be a string if empty values
+    * @param string $fieldname how is named the current date field
+    *                         (used to specify creation date or last update)
+    *
+    * @return string
+    */
+   static function dates($values = "", string $fieldname = 'dates'): string {
+      if (is_string($values) && strlen($values) == 0) {
```
Comment (trasher) [inline inc/dashboard/filter.class.php:78]: `$value` is reset if it has a string value, is it what you expect?

File: inc/dashboard/widget.class.php [hunk 3]
```diff
@@ -233,6 +294,13 @@ class="card big-number"
    }
 
 
+   public static function summaryNumber(array $params = []): string {
+      $params['class'] = 'summary-numbers';
+      //$params['label'] = '';
```
Comment (trasher) [inline inc/dashboard/widget.class.php:299]:
```suggestion
```

File: inc/ticket.class.php [hunk 4]
```diff
@@ -7324,6 +7324,128 @@ public static function merge(int $merge_target_id, array $ticket_ids, array &$st
    }
 
 
+   /**
+    * Check profiles and detect where criteria from existing rights
+    *
+    * @return array criteria to apply to an iterator query
+    */
+   public static function getCriteriaFromProfile() {
```
Comment (trasher) [inline inc/ticket.class.php:7332]: Should be unit tested

File: inc/dashboard/filter.class.php [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+         $values = [];
+      }
```
Comment (trasher) [inline inc/dashboard/filter.class.php:80]:
```suggestion
      if (is_string($values)) {
         $values = (strlen($values) == 0 ? [] : [$values]);
      }
```

File: inc/dashboard/filter.class.php (hunk 2 above)
Comment (trasher) [reply inc/dashboard/filter.class.php:87]: see https://github.com/glpi-project/glpi/pull/7793#discussion_r494839862

File: inc/ticket.class.php (hunk 4 above)
Comment (trasher) [reply inc/ticket.class.php:7332]: (not a blocker)

File: inc/dashboard/widget.class.php (hunk 3 above)
Comment (orthagh) [reply inc/dashboard/widget.class.php:299]: it can be an array (see docblock), i think mixed type is not available with our php version requirement

Comment (orthagh) [reply inc/dashboard/widget.class.php:299]: the fact is:
- when we have an empty value , we receive ""
- when we have a filled value, we receive ["date1", "date2]

Comment (trasher) [reply inc/dashboard/widget.class.php:299]: heu... it's just about removing a commented line here :)

Comment (orthagh) [reply inc/dashboard/widget.class.php:299]: wrong place, it was for the previous entry, sorry

Comment (trasher) [reply inc/dashboard/widget.class.php:299]: np, so `$values` is either an array or an empty string, therefore whecking its lenght is not required

File: inc/dashboard/filter.class.php (hunk 5 above)
Comment (trasher) [reply inc/dashboard/filter.class.php:89]: not ok

File: inc/dashboard/widget.class.php (hunk 3 above)
Comment (orthagh) [reply inc/dashboard/widget.class.php:299]: yes, commit is coming

**Review by trasher — APPROVED** [review]

File: inc/dashboard/filter.class.php [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    *
+    * @return string
+    */
+   static function dates_mod($values): string {
+      return self::dates($values, "dates_mod");
+   }
+
+
+   static function itilcategory(string $value = "0"): string {
+      return self::dropdown($value, 'itilcategory', "\ItilCategory");
```
Comment (cedric-anne) [inline inc/dashboard/filter.class.php:133]:
```suggestion
      return self::dropdown($value, 'itilcategory', ITILCategory::class);
```

File: inc/dashboard/filter.class.php [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+   }
+
+   static function requesttype(string $value = "0"): string {
+      return self::dropdown($value, 'requesttype', "\RequestType");
```
Comment (cedric-anne) [inline inc/dashboard/filter.class.php:137]:
```suggestion
      return self::dropdown($value, 'requesttype', RequestType::class);
```

File: inc/dashboard/filter.class.php [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+   }
+
+   static function location(string $value = "0"): string {
+      return self::dropdown($value, 'location', "\Location");
```
Comment (cedric-anne) [inline inc/dashboard/filter.class.php:141]:
```suggestion
      return self::dropdown($value, 'location', Location::class);
```

File: inc/dashboard/filter.class.php [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+   }
+
+   static function manufacturer(string $value = "0"): string {
+      return self::dropdown($value, 'manufacturer', "\Manufacturer");
```
Comment (cedric-anne) [inline inc/dashboard/filter.class.php:145]:
```suggestion
      return self::dropdown($value, 'manufacturer', Manufacturer::class);
```

File: js/dashboard.js [hunk 10]
```diff
@@ -106,7 +108,13 @@ var Dashboard = {
       // retieve cards content by ajax
       if (Dashboard.ajax_cards) {
          Dashboard.getCardsAjax();
-      } else {
+      }
+
+      // init filters from storage
+      Dashboard.initFilters();
```
Comment (cedric-anne) [inline js/dashboard.js:114]: I would have done that before `Dashboard.getCardsAjax();`.

Comment (orthagh) [reply js/dashboard.js]: You're completely right

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**